src/app/ReadClient.cpp

```diff
@@ -42,20 +84,34 @@ ReadClient::ReadClient(InteractionModelEngine * apImEngine, Messaging::ExchangeM
 
     mpImEngine = apImEngine;
 
+    ResubscribeClear();
+
     if (aInteractionType == InteractionType::Subscribe)
     {
         mpImEngine->AddReadClient(this);
     }
 }
 
+void ReadClient::ResubscribeClear()
```

Comment:
Should this be called something like "WaitForResubscribe()"?  Or "ClearActiveSubscriptionState()"?

Reply:
update it with ClearActiveSubscriptionState. thanks